axum/src/middleware/mod.rs

```diff
@@ -5,3 +5,9 @@
 mod from_fn;
 
 pub use self::from_fn::{from_fn, FromFn, FromFnLayer, Next};
+
+pub mod future {
+    //! Future types.
+
+    pub use super::from_fn::ResponseFuture as FromFnResponseFuture;
+}
```

Comment:
Its rare that users actually care about the future type so tower relates crates commonly have a `futures` module like this. [Prior art: `tower::util::future`](https://docs.rs/tower/latest/tower/util/future/index.html).